src/test/StkAaveRetrievalE2E.t.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity ^0.8.0;
+
+// testing libraries
+import "@forge-std/Test.sol";
+
+// contract dependencies
+import "../external/aave/IStaticATokenLM.sol";
+import "../external/aave/IAaveIncentivesController.sol";
+import {GovHelpers} from "@aave-helpers/GovHelpers.sol";
+import {AaveV2Ethereum} from "@aave-address-book/AaveV2Ethereum.sol";
+import {StkAaveRetrieval} from "../StkAaveRetrieval.sol";
+import {ProposalPayload} from "../ProposalPayload.sol";
+import {DeployMainnetProposal} from "../../script/DeployMainnetProposal.s.sol";
+import "@openzeppelin/token/ERC20/IERC20.sol";
+
+contract StkAaveRetrievalE2ETest is Test {
+    address public constant AAVE_WHALE = 0xBE0eB53F46cd790Cd13851d5EFf43D12404d33E8;
+
+    uint256 public proposalId;
+
+    StkAaveRetrieval private stkAaveRetrieval;
+    ProposalPayload private proposalPayload;
+
+    IERC20 STK_AAVE = IERC20(0x4da27a545c0c5B758a6BA100e3a049001de870f5);
+
+    address public incentivesControllerAddr;
+    address public balancerDAO;
+    address public balancerMultisig;
+
+    IStaticATokenLM public wrapped_aDAI;
+    IStaticATokenLM public wrapped_aUSDC;
+    IStaticATokenLM public wrapped_aUSDT;
+
+    function setUp() public {
+        // To fork at a specific block: vm.createSelectFork(vm.rpcUrl("mainnet", BLOCK_NUMBER));
+        vm.createSelectFork(vm.rpcUrl("mainnet"));
+
+        stkAaveRetrieval = new StkAaveRetrieval();
+        incentivesControllerAddr = stkAaveRetrieval.INCENTIVES_CONTROLLER();
+        balancerDAO = stkAaveRetrieval.BALANCER_DAO();
+        balancerMultisig = stkAaveRetrieval.BALANCER_MULTISIG();
+        wrapped_aDAI = IStaticATokenLM(stkAaveRetrieval.WRAPPED_ADAI());
+        wrapped_aUSDC = IStaticATokenLM(stkAaveRetrieval.WRAPPED_AUSDC());
+        wrapped_aUSDT = IStaticATokenLM(stkAaveRetrieval.WRAPPED_AUSDT());
+        // Deploy Payload
+        ProposalPayload proposalPayload = new ProposalPayload(stkAaveRetrieval);
+
+        // Create Proposal
+        vm.prank(AAVE_WHALE);
+        proposalId = DeployMainnetProposal._deployMainnetProposal(
+            address(proposalPayload),
+            0x344d3181f08b3186228b93bac0005a3a961238164b8b06cbb5f0428a9180b8a7 // TODO: Replace with actual IPFS Hash
+        );
+    }
+
+    function testClaimerUnsetPreProposal() public {
+        // Call retrieve() on StkAaveRetrieval contract as Balancer multisig without the Proposal being executed
+        vm.prank(balancerMultisig);
+        vm.expectRevert(bytes("Contract not set as claimer"));
+        stkAaveRetrieval.retrieve();
+    }
+
+    function testRetrieveNotBalancerMultisigPreProposal() public {
+        // Call retrieve() on StkAaveRetrieval contract without the Proposal being executed
+        // the msg.sender will be this contract, which is NOT the balancer multisig
+        GovHelpers.passVoteAndExecute(vm, proposalId);
+        vm.expectRevert(bytes("Only Balancer Multisig"));
+        stkAaveRetrieval.retrieve();
+    }
```

Comment:
This is great! Can you add another test for checking NotBalancerMultisig Post proposal as well?

Reply:
Just added. Turns out this one was actually post proposal, so I added pre proposal as well which is the same thing without line 67